avatars/animationHelpers.js

```diff
@@ -1249,17 +1253,49 @@ export const _applyAnimation = (avatar, now, moveFactors, timeDiffS) => {
       const f = MathUtils.clamp(swimTimeS / 0.2, 0, 1);
       // if (isPosition) console.log(f);
 
+      
+
       if (!isPosition) {
         localQuaternion2.fromArray(v2);
         localQuaternion3.fromArray(v3);
         localQuaternion4.fromArray(v4);
+        localQuaternion5.fromArray([0.3826834, 0, 0, 0.9238795]); // x: 45deg
+        localQuaternion6.fromArray([0,0,0,1]);
+        localQuaternion7.fromArray([1,0,0,0]);
+        
+        
         // // can't use idleWalkFactor & walkRunFactor here, otherwise "Impulsive breaststroke swim animation" will turn into "freestyle animation" when speed is fast,
         // // and will turn into "floating" a little when speed is slow.
         // localQuaternion3.slerp(localQuaternion4, walkRunFactor);
         // localQuaternion2.slerp(localQuaternion3, idleWalkFactor);
         localQuaternion3.slerp(localQuaternion4, avatar.sprintFactor);
         localQuaternion2.slerp(localQuaternion3, avatar.movementsTransitionFactor);
+
+        
+
+        // Swinging Hips for Breaststroke
+        if(boneName === 'Hips' && avatar.sprintFactor === 0 && movementsTimeS > 0) {
```

Comment:
since we are checking if bonename === hips below, could we embed this boolean check inside the other one?